Sources/Build/PackageVersion.swift

```diff
@@ -15,7 +15,7 @@ import class Utility.Git
 import struct Utility.Path
 
 public func generateVersionData(_ rootDir: AbsolutePath, rootPackage: Package, externalPackages: [Package]) throws {
-    let dirPath = rootDir.appending(".build/versionData")
+    let dirPath = rootDir.appending(components: ".build", "versionData")
```

Comment:
Great that we agree.  What I found particularly readable was `let path = rootDir + ".build" + "versionData" + moduleName` etc, but I suspect we'll agree to disagree about that one.

However:  this does raise the question about whether RelativePath should also have a `.appending(component:)` method.  It probably needs one (sorry if I missed it, I didn't see one in the diffs).